draft-ietf-tls-extended-key-update.md

```diff
@@ -534,37 +440,54 @@ Auth | {CertificateVerify}
   (with key_share))]
                                <-------- [ExtendedKeyUpdate(response
                                            (accepted, with key_share))]
- [ExtendedKeyUpdate(new_key_update)]     -------->
-                                         <--------
+										# Server derives new secrets
+# Client derives new secrets			
+ [ExtendedKeyUpdate(new_key_update)]
+                               -------->
+# Client updates SEND keys here
+                                    # Server updates RECEIVE keys here
+   							   <--------
                                     [ExtendedKeyUpdate(new_key_update)]
+									# Server updates SEND keys here
+
+# Client updates RECEIVE keys here
 ~~~
-{: #fig-key-update2 title="Extended Key Update Example Exchange."}
+{: #fig-key-update2 title="Extended Key Update Example."}
 
 #  DTLS 1.3 Considerations
 
 Unlike TLS 1.3, DTLS 1.3 implementations must take into account that handshake
 messages are not transmitted over a reliable transport protocol.
 
+Due to the possibility of an `ExtendedKeyUpdate(new_key_update)` message being
+lost and thereby preventing the sender of that message from updating its keying
+material, receivers MUST retain the pre-update keying material until receipt
+and successful decryption of a message using the new keys.
+
+Due to loss and/or reordering, DTLS 1.3 peers MAY receive a record with an
+older epoch than the current one. They SHOULD attempt to process such records
+for that epoch but MAY opt to discard such out-of-epoch records.
+
 The exchange has the following steps:
 
-1. The initiator sends an `ExtendedKeyUpdate(request)` message, which contains a key
-   share. While an extended key update is in progress, the initiator MUST NOT
+1. The initiator sends an `ExtendedKeyUpdate(request)` message, which contains a
+   key share. While an extended key update is in progress, the initiator MUST NOT
    initiate further key updates.
```

Comment:
```suggestion
   initiate further key updates. This message is sent reliably using DTLS handshake retransmission.
```

Reply:
The message per se is not sent reliably without the response message, which gives the sender an assurance that the request message has indeed arrived.

Reply:
we may want to add: This message is subject to DTLS handshake retransmission, but delivery is only
confirmed when the initiator receives the corresponding `ExtendedKeyUpdate(response)`.